terraform/modules/dandiset_bucket/main.tf

```diff
@@ -136,6 +136,42 @@ data "aws_iam_policy_document" "dandiset_bucket_owner" {
   }
 }
 
+resource "aws_s3_bucket_policy" "dandiset_embargoed_objects" {
+  count    = var.public ? 1 : 0
+  provider = aws
+  bucket   = aws_s3_bucket.dandiset_bucket.id
+  policy   = data.aws_iam_policy_document.dandiset_embargoed_objects[0].json
+}
+
+data "aws_iam_policy_document" "dandiset_embargoed_objects" {
+  count   = var.public ? 1 : 0
+  version = "2008-10-17"
+
+  statement {
+    effect = "Deny"
```

Comment:
Just wanted to point out that I like this approach more than an Allow-based approach, since [Denys always take priority over Allows](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_evaluation-logic.html#policy-eval-denyallow)